libraries/griptape_nodes_library/griptape_nodes_library/engine/delete_file.py

```diff
@@ -0,0 +1,534 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from enum import Enum
+from pathlib import Path
+from typing import TYPE_CHECKING, Any
+
+from griptape_nodes.exe_types.core_types import Parameter, ParameterMessage, ParameterMode
+from griptape_nodes.exe_types.node_types import SuccessFailureNode
+
+if TYPE_CHECKING:
+    from griptape_nodes.exe_types.node_types import BaseNode
+from griptape_nodes.retained_mode.events.os_events import (
+    DeleteFileRequest,
+    DeleteFileResultFailure,
+    DeleteFileResultSuccess,
+    GetFileInfoRequest,
+    GetFileInfoResultFailure,
+    GetFileInfoResultSuccess,
+    ListDirectoryRequest,
+    ListDirectoryResultFailure,
+    ListDirectoryResultSuccess,
+)
+from griptape_nodes.retained_mode.griptape_nodes import GriptapeNodes
+from griptape_nodes.traits.file_system_picker import FileSystemPicker
+
+
+class DeletionStatus(Enum):
+    """Status of a deletion attempt."""
+
+    PENDING = "pending"
+    SUCCESS = "success"
+    FAILED = "failed"
+    SKIPPED = "skipped"  # Skipped because parent directory was deleted
+    INVALID = "invalid"  # Invalid or inaccessible path
+
+
+@dataclass
+class DeleteFileInfo:
+    """Information about a file/directory deletion attempt."""
+
+    path: str  # Workspace-relative path (for portability)
+    is_directory: bool
+    absolute_path: str  # Absolute resolved path
+    status: DeletionStatus = DeletionStatus.PENDING
+    failure_reason: str | None = None
+    deleted_paths: list[str] = field(default_factory=list)  # Paths actually deleted (from OS result)
+    explicitly_requested: bool = False  # True if user specified this path, False if discovered via recursion
+
+
+class DeleteFile(SuccessFailureNode):
+    """Delete files and/or directories from the file system.
+
+    Directories are deleted with all their contents.
+    Accepts single path (str) or multiple paths (list[str]).
+    Supports glob patterns (e.g., "/path/to/*.txt") for matching multiple files.
+    """
+
+    def __init__(self, name: str, metadata: dict[Any, Any] | None = None) -> None:
+        super().__init__(name, metadata)
+
+        # Input parameter - accepts str or list[str]
+        self.file_paths = Parameter(
+            name="file_paths",
+            type="list",
+            allowed_modes={ParameterMode.INPUT, ParameterMode.PROPERTY},
+            input_types=["str", "list"],
+            default_value=[],
+            tooltip="Path(s) to file(s) or directory(ies) to delete. Supports glob patterns (e.g., '/path/*.txt').",
+        )
+        self.file_paths.add_trait(
```

Comment:
I don't think a file picker works on a list trait? 

wondering if this should be a ParameterList of type str? What do you think?